server/src/main/java/org/apache/druid/server/ClientQuerySegmentWalker.java

```diff
@@ -670,8 +671,14 @@ private static <T, QueryType extends Query<T>> DataSource toInlineDataSource(
         if (limitAccumulator.get() >= rowLimitToUse) {
           subqueryStatsProvider.incrementQueriesExceedingRowLimit();
           throw ResourceLimitExceededException.withMessage(
-              "Cannot issue the query, subqueries generated results beyond maximum[%d] rows",
-              rowLimitToUse
+              "Cannot issue the query, subqueries generated results beyond maximum[%d] rows. Try setting the"
+              + "%s in the query context to '%s' for enabling byte based limit, which chooses an optimal limit based on "
+              + "memory size and result's heap usage or manually configure the values of either %s or %s in the query context",
```

Comment:
We should also mention that this limit should be carefully set since it can be afftect other queries on the broker and can potentially lead to OOM.